inception-imls-stringmatch/src/main/java/de/tudarmstadt/ukp/inception/recommendation/imls/stringmatch/Trie.java

```diff
@@ -379,6 +419,30 @@ private void keys(final Character c, final Node n, final StringBuilder b,
         }
     }
 
+    public static interface KeySanitizerFactory
```

Comment:
I do not understand why you need a factory here. Also, I would move this static class to its own file. For me, nested classes/interfaces are only used inside a file, but you implement it outside.

Reply:
I need a factory because the actual sanitizer is stateful. It is a character stream processor which needs to have a memory about the last character(s) it has seen. Thus, I cannot re-use a sanitizer instance.